decode.go

```diff
@@ -96,7 +96,11 @@ type Decoded struct {
 	pos         int64
 }
 
-func (d *Decoded) readFrame() error {
+// Decoded is the old type name for the Decoder
+// DEPRECATED
+type Decoded Decoder
```

Comment:
Typing it like this still gives us the same methods as Decoder - am I missing something? Either way I am happy to change it if that's what you wish.